frame/contracts/src/benchmarking/code.rs

```diff
@@ -136,20 +171,69 @@ impl<T: Trait> From<ModuleDefinition> for WasmModule<T> {
 				.build()
 		}
 
-		let code = contract.build().to_bytes().unwrap();
+		// Add global variables
+		if def.num_globals > 0 {
+			use rand::{prelude::*, distributions::Standard};
+			let rng = rand_pcg::Pcg32::seed_from_u64(3112244599778833558);
+			for val in rng.sample_iter(Standard).take(def.num_globals as usize) {
+				contract = contract
+					.global()
+					.value_type().i64()
+					.mutable()
+					.init_expr(Instruction::I64Const(val))
+					.build()
+			}
+		}
+
+		// Add function pointer table
+		if let Some(table) = def.table {
+			contract = contract
+				.table()
+				.with_min(table.num_elements)
+				.with_max(Some(table.num_elements))
+				.with_element(0, vec![table.function_index; table.num_elements as usize])
+				.build();
+		}
+
+		let mut code = contract.build();
+
+		// Inject stack height metering
+		if def.inject_stack_metering {
+			code = inject_limiter(
+				code,
+				Contracts::<T>::current_schedule().limits.stack_height
+			)
+			.unwrap();
+		}
+
+		let code = code.to_bytes().unwrap();
 		let hash = T::Hashing::hash(&code);
 		Self {
 			code,
-			hash
+			hash,
+			memory: def.memory,
 		}
 	}
 }
 
 impl<T: Trait> WasmModule<T> {
+	/// Creates a wasm module with an empty `call` and `deploy` function and nothing else.
 	pub fn dummy() -> Self {
 		ModuleDefinition::default().into()
 	}
 
+	/// Same as `dummy` but with maximum sized linear memory.
+	pub fn dummy_with_mem() -> Self {
+		ModuleDefinition {
+			memory: Some(ImportedMemory::max::<T>()),
+			.. Default::default()
+		}
+		.into()
+	}
+
+	/// Creates a wasm module of `target_bytes` size. Used to benchmark the performance of
+	/// `put_code` for different sizes of wasm modules. The generated module maximizes
+	/// instrumentation runtime by nesting blocks as deeply as possible given the byte budget.
```

Comment:
are those ending in html docs somewhere?

if so, better to decorate with header separated with empty line

```suggestion
	/// Create a wasm module of `target_bytes` size. 
	
	/// Used to benchmark the performance of
	/// `put_code` for different sizes of wasm modules. The generated module maximizes
	/// instrumentation runtime by nesting blocks as deeply as possible given the byte budget.
```

Reply:
The whole benchmarking module is private and guarded behind the `runtime-benchmarks` feature flag. It is extremely unlikely that anyone will consume those comments using html.